### PR TITLE
core: add signed data type for BeaconCommitteeSubscription

### PR DESCRIPTION
**Pull request by dB2510**
Adds DutyPrepareAggregator and signed data type for BeaconCommitteeSubscription.

category: feature
ticket: #1064

### CODE REVIEW

File: core/signeddata.go [hunk 1]
```diff
@@ -691,6 +692,56 @@ func (s SignedRandao) clone() (SignedRandao, error) {
 	return resp, nil
 }
 
+// NewPartialSignedBeaconCommitteeSubscription is a convenience function which returns new partially signed BeaconCommitteeSubscription.
+func NewPartialSignedBeaconCommitteeSubscription(sub eth2exp.BeaconCommitteeSubscription, shareIdx int) ParSignedData {
```
Comment (corverroos) [inline core/signeddata.go:696]: wondering if we can drop the `Beacon` part without loss of context?

File: core/signeddata_test.go [hunk 2]
```diff
@@ -67,3 +68,20 @@ func TestSetVersionedValidatorRegistrationSig(t *testing.T) {
 	require.NoError(t, err)
 	require.NotEqual(t, clone.Signature(), registration.Signature())
 }
+
+func TestSetBeaconCommitteeSubscription(t *testing.T) {
```
Comment (corverroos) [inline core/signeddata_test.go:72]: can this be refactored into a table test?

**Review by corverroos — APPROVED** [review]

File: core/signeddata.go (hunk 1 above)
Comment (dB2510) [reply core/signeddata.go:696]: i suggest we should keep this to align with underlying type

File: core/signeddata_test.go (hunk 2 above)
Comment (dB2510) [reply core/signeddata_test.go]: done

**Pull request merged**